### PR TITLE
[test] add retain_active_session option for ePSKc deactivate API

### PR DESCRIPTION
**Pull request by mia1yang**
Depends-On: openthread/ot-br-posix#2537

### CODE REVIEW

**Review by bukepo — APPROVED** [review]

File: tests/scripts/thread-cert/node.py [hunk 1]
```diff
@@ -425,8 +425,8 @@ def nat64_set_enabled(self, enable):
     def activate_ephemeral_key_mode(self, lifetime):
         return self.call_dbus_method('io.openthread.BorderRouter', 'ActivateEphemeralKeyMode', lifetime)
 
-    def deactivate_ephemeral_key_mode(self):
-        return self.call_dbus_method('io.openthread.BorderRouter', 'DeactivateEphemeralKeyMode')
+    def deactivate_ephemeral_key_mode(self, retain_active_session):
```
Comment (bukepo) [inline tests/scripts/thread-cert/node.py:428]:
```suggestion
    def deactivate_ephemeral_key_mode(self, retain_active_session=False):
```

File: tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py [hunk 2]
```diff
@@ -111,7 +111,8 @@ def test(self):
         self.check_meshcop_e_service(host, True)
 
         # deactivate ePSKc mode
-        br1.deactivate_ephemeral_key_mode()
+        retain_active_session = False
```
Comment (sunytt) [inline tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py:114]: also test the case when `retain_active_session = True` ?

Comment (mia1yang) [reply tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py]: Add another pair of activate/deactivate for different `retain_active_session` options. Thanks~

**Review by sunytt — APPROVED** [review]
LGTM

File: tests/scripts/thread-cert/node.py (hunk 1 above)
Comment (mia1yang) [reply tests/scripts/thread-cert/node.py:428]: Thanks, accepted.

File: tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py [hunk 3]
```diff
@@ -110,8 +110,23 @@ def test(self):
         # check Meshcop-e service
         self.check_meshcop_e_service(host, True)
 
-        # deactivate ePSKc mode
-        br1.deactivate_ephemeral_key_mode()
+        # deactivate ePSKc mode in force
+        br1.deactivate_ephemeral_key_mode(retain_active_session=False)
```
Comment (librasungirl) [inline tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py:114]: probably in separate PR: introduce ot-commissioner host and verify the scenarios when there is DTLS secure session established

**Review by librasungirl — APPROVED** [review]

**Comment by mia1yang** [comment]
This PR is the unit test for ot-br-posix PR https://github.com/openthread/ot-br-posix/pull/2537; After this PR merged,  #2537 CI failures would be clear (due to DBus parameter added and checked)

**Pull request merged**